app/bundles/EmailBundle/Tests/Model/SendEmailToContactTest.php

```diff
@@ -725,4 +755,98 @@ function () use ($emailMock) {
+    {
+        // Use our test token transport limiting to 1 recipient per queue
+        $transport = new BatchTransport(true, 1);
+        $mailer    = new \Swift_Mailer($transport);
+
+        // Mock factory to ensure that queue mode is handled until MailHelper is refactored completely away from MauticFactory
+
+        $this->factoryMock->expects($this->at(3))->method('getParameter')->with('mailer_spool_type')->willReturn($type);
+        $this->factoryMock->expects($this->at(1))->method('getParameter')->willReturn('');
+
+        $this->factoryMock->method('getLogger')
+            ->willReturn(
+                new NullLogger()
+            );
+
+        $this->factoryMock->method('getEntityManager')
+            ->willReturn($this->mockEm);
+
+        $this->mockDispatcher->method('dispatch')
+            ->willReturnCallback(
+                function ($eventName, EmailSendEvent $event) {
+                    $lead = $event->getLead();
+
+                    $tokens = [];
+                    foreach ($lead as $field => $value) {
+                        $tokens["{contactfield=$field}"] = $value;
+                    }
+                    $tokens['{hash}'] = $event->getIdHash();
+
+                    $event->addTokens($tokens);
+                }
+            );
+        $this->factoryMock->method('getDispatcher')
+            ->willReturn($this->mockDispatcher);
+
+        $this->factoryMock->method('getRouter')
+            ->willReturn($this->routerMock);
+
+        $this->emailModelMock->method('getCopyRepository')
+            ->willReturn($this->copyRepoMock);
+
+        $this->factoryMock->method('getModel')
+            ->willReturn($this->emailModelMock);
+
+        $mailHelper = $this->getMockBuilder(MailHelper::class)
+            ->setConstructorArgs([$this->factoryMock, $mailer])
+            ->setMethods(null)
+            ->getMock();
+
+        // Enable queueing
+        $mailHelper->enableQueue();
+
+        $this->statRepository->method('saveEntity')
+            ->willReturnCallback(
+                function (Stat $stat) {
+                    $tokens = $stat->getTokens();
+                    $this->assertGreaterThan(1, count($tokens));
+                    $this->assertEquals($stat->getTrackingHash(), $tokens['{hash}']);
+                }
+            );
+
+        $translator = $this->createMock(Translator::class);
+
+        $statHelper = new StatHelper($this->statRepository);
+
+        $this->builder->expects($this->any())
+            ->method('getOptions')
+            ->will($this->returnValue(['with_immediately' => true]));
+
+        $model = new SendEmailToContact($mailHelper, $statHelper, $this->dncModel, $translator);
+        $model->setSampleMailer();
+        $this->assertNotNull($model->getTemporaryMailer());
+
+        $model->reset();
+        $this->assertNull($model->getTemporaryMailer());
+
+        $this->builder->expects($this->any())
```

Comment:
The method for `getOptions` is set on line 830. This will take no effect. You need to split this test to 2 separate tests.
Also, please move it to a new class, this test is pretty huge.